### ISSUE TITLE
A new option for CLI method to make automatic backup and close firefox

### ISSUE THREAD

**Issue by schmurtzm** [gildas-lormeau/SingleFile#546]
**Is your feature request related to a problem? Please describe.**
The cli method require skill and best option is a docker which is not very simple or portable.

**Describe the solution you'd like**
An option to use the extension with command line in a simple way. The extension [save-page-we](https://addons.mozilla.org/fr/firefox/addon/save-page-we/) has a great idea : 

```
If this option is enabled:

• When Firefox starts, Save Page WE will be automatically initiated using the current Button Action.
• When the save operation is complete, Firefox will be automatically closed.

To exit automation mode, immediately after Firefox starts up, use the 'Cancel Save' menu command or keyboard shortcut to cancel the save operation, and then disable the option to use automation.

Note, the new saved method is always used when the use automation option is enabled.
```

And BTW, thanks for this great tool ;)

**Comment by gildas-lormeau** [comment]
I'm sorry but I don't think the CLI tool is so hard to install. I'm also not fond of copying what Save Page WE does to be honest.

**Comment by gildas-lormeau** [comment]
Actually, I might be wrong but isn't it possible to achieve the same result by enabling auto-save for all tabs and the checking the option `User interface > auto-close the tab after the page is saved` in SingleFile?

**Comment by schmurtzm** [comment]
> I'm sorry but I don't think the CLI tool is so hard to install. 
>
??? you're too immersed in the geek world  😉 
> I'm also not fond of copying what Save Page WE does to be honest.
>
I don't get it. We speak about a new feature... You implement only what doesn't exist on this world ? 😄 
> Actually, I might be wrong but isn't it possible to achieve the same result by enabling the auto-save for all tabs and checking the option `User interface > auto-close the tab after the page is saved` in SingleFile?
>
All these "No!" for a feature that you have already implemented 😁 
I'm kidding you a little, thank you finally for this answer, I'm going to test it tonight. Don't close it immediatly, I will report if it work fine because it can be a good alternative to the CLI method ;)

**Comment by gildas-lormeau** [comment]
@schmurtzm I'm sorry, however the good news is that I think my proposal (almost) works.

I did the test with Chrome and I noticed there was a bug. The browser was closed too soon and the page was not saved. I fixed this bug in the version 1.18.44 that I just uploaded. Hopefully, it should be available for download tomorrow. Meanwhile, you can do some tests with the version 0.6.29 of SingleFileZ if you want.

**Comment by schmurtzm** [comment]
OK I've made 2 tests : 
`----------------------------------------`
First I've installed the CLI tool to try. On run I had this error message  : 
    &nbsp;&nbsp;"Failed to launch the browser process! spawn chrome ENOENT"

(may be due to my tests with puppeteer-cli before ? I don't know...)
I find this parameter in some issues : "--browser-executable-path" and it works when I specify the chromium path from puppeteer-cli  (I don't have another chromium installed...). It seems that there is a problem with chromium installation...

After that it works great. There is something bad in the npm install script or this is just that I'm zero with nodejs 😅  ? 
My next step will be to try to make something portable with nodejs version.
`----------------------------------------`
Then I took a firefox nightly portable and I've tried the last version of SingleFile (1.18.44) on it , the auto-close / auto save works fine with the last github version (otherwise like you said the navigator is closed to soon and nothing is saved with the 1.18.42.)

To finish I've modified the portable app config file to allow multiple instance of firefox (because I always have my main firefox launched) and I run it with the "-headless" parameter : perfect portable tool to backup web page ! 
`----------------------------------------`

The initial request of this issue is accomplished, feel free to close it, happy of the result 👍 
Thanks for your advice !

<sub>ps: I didn't try with SingleFileZ but I don't like too much the fact that the extension is necessary to read the backuped file.<sub>

**Comment by gildas-lormeau** [comment]
Thank you for thee feedback. If Chrome is not installed in the default folder, you have indeed to use the `----browser-executable-path` switch.

Note that you need to install SingleFileZ to view saved pages if and only if:
- you're not using Firefox
- you're opening the page from the filesystem

For example, you can view this page in any modern browser without installing anything: https://gildas-lormeau.github.io/.
